### PR TITLE
Update version

### LINKED ISSUE #65: Update to latest sway 0.41

### PR DESCRIPTION
**Pull request by Braqzen**
Closes #65 

Note: CI passes but I'm unsure if the workspace is working correctly because locally running forc ignores the pinned versions.

### CODE REVIEW

**Comment by Voxelot** [comment]
versions to target: 

- fuel-core v0.18.3
- fuels-rs v0.43.0
- sway v0.41.0

**Comment by xgreenx** [comment]
I want to highlight that the code in this repository and https://github.com/FuelLabs/bridge-message-predicates/pull/26 have many of the same parts. Maybe it is worth to re-work the current structure of the bridge. Because we do the same update across several repositories

**Comment by Braqzen** [comment]
> I want to highlight that the code in this repository and [FuelLabs/bridge-message-predicates#26](https://github.com/FuelLabs/bridge-message-predicates/pull/26) have many of the same parts. Maybe it is worth to re-work the current structure of the bridge. Because we do the same update across several repositories

I'm aware of multiple repositories for the bridge but I'm unaware of why they are split up the way they are.
Perhaps we should create a monorepo for the various components?

File: bridge-fungible-token/tests/harness.rs [hunk 1]
```diff
@@ -235,12 +234,12 @@ mod success {
                 Bits256::from_hex_str(FROM).unwrap(),
                 Bits256::from_hex_str(BRIDGED_TOKEN).unwrap(),
             )
-            .append_message_outputs(1)
+            .tx_params(TxParameters::default().set_gas_limit(2000000))
```
Comment (xgreenx) [inline bridge-fungible-token/tests/harness.rs:237]: Do you really need to specify gas limit here?

File: bridge-fungible-token/src/utils.sw [hunk 2]
```diff
@@ -150,28 +150,20 @@ pub fn decompose(val: b256) -> (u64, u64, u64, u64) {
...
-    // msg_data.token = input_message_data(msg_idx, 32).into();
-    let from = input_message_data(msg_idx, 32 + 32);
-    let ptr_2 = __addr_of(msg_data.from);
-    from.buf.ptr().copy_to::<b256>(ptr_2, 1);
-    // msg_data.from = input_message_data(msg_idx, 32 + 32).into();
-    let amount = input_message_data(msg_idx, 32 + 32 + 32 + 32);
-    let ptr_3 = __addr_of(msg_data.amount);
-    amount.buf.ptr().copy_to::<b256>(ptr_3, 1);
-    // msg_data.amount = input_message_data(msg_idx, 32 + 32 + 32 + 32).into();
+    // TODO: Bug, have to mutate this struct for these values or tests fail
```
Comment (xgreenx) [inline bridge-fungible-token/src/utils.sw:164]: Do we still need this part? 
image

File: README.md [hunk 3]
```diff
@@ -6,44 +15,21 @@ The contract responsible for sending/receiving messages from the base layer gate
...
-Run tests:
+## Quickstart
 
-```sh
-cargo test
-```
+### Building Sway
 
-## Contributing
+In the root of the repository run `forc build` to build all the Sway programs. Each Sway project uses a `fuel-toolchain.toml` file with pinned versions so `forc` will install the correct versions for you if you do not have them installed.
```
Comment (luizstacio) [inline README.md:28]: I think, breaking the sections in lines is better to read than adding the commands in the middle of the text.

File: README.md [hunk 4]
```diff
@@ -6,43 +15,30 @@ The contract responsible for sending/receiving messages from the base layer gate
...
 
-Run tests:
+### Running Rust Tests
 
-```sh
-cargo test
-```
+After the Sway programs have been built run the following command in the root of the repository.
 
-## Contributing
```
Comment (pixelcircuits) [inline README.md:39]: Why was this section removed?

Comment (Braqzen) [reply README.md:39]: It doesn't add any value aside from telling you to format code and CI will fail if it's not formatted anyway so they have to format

Comment (pixelcircuits) [reply README.md:39]: Right, I thought the whole point of having it in the readme is so you don't have to wait until CI finishes to see that you had to run the formatter. This avoids unnecessary "formatting" commits that ruin the last commit message preview in the github code explorer. Has the team been removing the "contributing" section in other repos too?

File: bridge-fungible-token/tests/utils/environment.rs [hunk 5]
```diff
@@ -1,33 +1,55 @@
...
+    "../bridge-fungible-token/out/debug/bridge_fungible_token.bin";
+const DEPOSIT_RECIPIENT_CONTRACT_BINARY: &str =
+    "../test-deposit-recipient-contract/out/debug/test_deposit_recipient_contract.bin";
+
+abigen!(
+    Contract(
+        name = "BridgeFungibleTokenContract",
+        abi = "./bridge-fungible-token/out/debug/bridge_fungible_token-abi.json",
+    ),
+    Predicate(
```
Comment (pixelcircuits) [inline bridge-fungible-token/tests/utils/environment.rs:39]: This can be a separate issue but I just wanted to mention that we can remove predicate and script code now by including a dependency for the [predicate repo crate](https://crates.io/crates/fuel-contract-message-predicate). Then whenever we need the the bytecode for either the script or predicate you just use something like `fuel_contract_message_predicate::script_bytecode()`. The crate also provides functions for getting the predicate root quickly too (`fuel_contract_message_predicate::predicate_root`).

File: docs/deploy_docs.md [hunk 6]
```diff
@@ -2,13 +2,6 @@
 
 Each token to be bridged needs to have a corresponding bridge fungible token contract defined on the Fuel chain. These contracts handle messaging with the base layer gateway contract and are responsible for minting and burning the Fuel side token proxies.
 
-## Dependencies
```
Comment (pixelcircuits) [inline docs/deploy_docs.md:5]: Why was this removed? Shouldn't there be some easy reference for developers? Especially since to run `forc-deploy` you need both of these dependencies.

**Review by pixelcircuits — COMMENTED** [review]
I did a quick sweep through all changes and everything looked fine, although a lot of shifting things around. I then did a more thorough review of the sway code to make sure security is maintained and it looks good. My only comments are stuff that are probably out of scope for this issue, although I would suggest removing some of the bigger changes made to the documentation. I think major doc updates might be out of scope for this issue anyway. Might want to make it a separate issue and part of a larger effort to unify documentation standards across all the orgs repos.

File: README.md (hunk 4 above)
Comment (Braqzen) [reply README.md:39]: I believe most repos have squashed commits so this shouldn't be an issue.
Once you learn to format your code once you do it everywhere. Having to remember to check the readme or have it enforced by the CI doesn't change anything and since we enforce it via CI it's redundant to remind people.

It's not about removing that section from places it's just that this content doesn't add value here so it was removed

File: bridge-fungible-token/tests/utils/environment.rs (hunk 5 above)
Comment (Braqzen) [reply bridge-fungible-token/tests/utils/environment.rs:39]: I don't know what will happen to that crate as we move towards a monorepo. Either way, I've raised an issue to refactor the repo prior to that merge so I'll try that in the subsequent issue

Comment (luizstacio) [reply bridge-fungible-token/tests/utils/environment.rs:39]: https://github.com/FuelLabs/fuels-rs/blob/6efa6f11c8f69f4ef604b70bb78d55f6d4bf76e1/Cargo.toml#L56

File: README.md (hunk 4 above)
Comment (luizstacio) [reply README.md:39]: I think is better to have the format command than don't have it. It don't hurt to help the devs here.

Comment (Braqzen) [reply README.md:39]: I disagree and that doesn't counter my point; however, since it's not a change that is harmful and there's 2 people that are requesting it now then I'll add it back in soon

**Review by luizstacio — APPROVED** [review]
